src/Tokenizer/Token.php

```diff
@@ -87,7 +87,7 @@ public function __construct($token)
         } else {
             throw new \InvalidArgumentException(sprintf(
                 'Cannot recognize input value as valid Token prototype, got "%s".',
-                \is_object($token) ? \get_class($token) : \gettype($token)
+                \get_class($token)
```

Comment:
Haha, I've tried the same and got the same results you can see in failed test.

In PHPDoc it is stated `$token` is array or string, so PHPStan claims this `else` is unreachable, when in fact it is. We will be able typehint this only in PHP 8+.